### ISSUE TITLE
Approach to downloading snapshot seems flawed

### ISSUE THREAD

**Issue by MonsieurNicolas** [stellar/stellar-core#1613]
It seems there is a problem with the way we trigger history catchup:
from my observations, nodes try to download checkpoints way too early (pointing to a bug in how we compute the amount of time we should wait before even trying to download the snapshot).

This in turn creates delays as the downloads are failing for no good reason (and is confusing operators).

I think it would make a lot more sense to not do it based on time but on the ledgers that we are observing from consensus (we can still give an ETA based on how far currently observed ledger is from the next checkpoint):
as we are already queuing up those ledgers, we could simply wait to trigger catchup when a ledger observed meets the "checkpoint criteria" instead of blindly starting the catchup process.

Basically we would change `LedgerManagerImpl::startCatchUp` to not always set the state to `LM_CATCHING_UP_STATE` but instead move to that state if `catchupHistory` can start work (ie `configuration` is on a snapshot boundary).
